src/MSBuild.Abstractions/ProjectItemComparer.cs

```diff
@@ -15,8 +15,23 @@ private ProjectItemComparer(bool compareMetadata)
             _compareMetadata = compareMetadata;
         }
 
-        public bool Equals(IProjectItem x, IProjectItem y)
+        public bool Equals(IProjectItem? x, IProjectItem? y)
```

Comment:
You can keep these non-nullable and add `[AllowNull]` in front of each to indicate that the intent is that you not pass null

Reply:
Yeah that’s preferable